improve debug struct UI

carletex opened the pull request:
First iteration on #725 

### 1. Fix padding 
This odd padding won't let me sleep haha (also making sure the inner structs padding looks good too)
![304829304-0d18fc28-b681-48f1-bf40-1f7644d62207](https://github.com/scaffold-eth/scaffold-eth-2/assets/2486142/f92354ea-417a-491b-8b77-d1a4cb001a96)

Now: 
![image](https://github.com/scaffold-eth/scaffold-eth-2/assets/2486142/4569fd88-8047-447b-8558-53442d47e349)

As Shiv mentioned, maybe it's a good idea to have a label on the value input too.

### Fix accordion style
In the current version, the text felt a bit "lost"
![304828797-4416cf52-a63f-4030-b983-49025fac8f5e](https://github.com/scaffold-eth/scaffold-eth-2/assets/2486142/2f3ca58b-4141-4a74-99cc-5f3da5b108f9)

Made a couple of tweaks:
1. add some background
![image](https://github.com/scaffold-eth/scaffold-eth-2/assets/2486142/429ec4bc-833c-4de0-b7f5-21a5add3a35c)
2. Add a border. If not, the inner struct will have the same problem (its background is the same as the parent bg). A border fixed it
![image](https://github.com/scaffold-eth/scaffold-eth-2/assets/2486142/0af16c29-0c3e-4905-a0ff-03f5150a7493)

----

We can add more stuff here, or keep iterating in another PR.

rin-st reviewed (approved):
looks much better!

Pabl0cks commented on packages/nextjs/app/debug/_components/contract/TupleArray.tsx:
```diff
@@ -97,12 +97,12 @@ export const TupleArray = ({ abiTupleParameter, setParentForm, parentStateObject
   return (
     <div>
-      <div className="collapse collapse-arrow">
+      <div className="collapse collapse-arrow bg-base-200 pl-2 py-1.5 mt-1 border-2 border-secondary">
```
Should we delete `mt-1` to have the same spacing between type text and border as regular inputs?

Pabl0cks commented on packages/nextjs/app/debug/_components/contract/Tuple.tsx:
```diff
@@ -27,12 +27,12 @@ export const Tuple = ({ abiTupleParameter, setParentForm, parentStateObjectKey }
   return (
     <div>
-      <div className="collapse collapse-arrow">
+      <div className="collapse collapse-arrow bg-base-200 pl-2 py-1.5 mt-1 border-2 border-secondary">
```
Should we delete `mt-1` to have the same spacing between type text and border as regular inputs?

If feels too close after deleting we could increase `ContractInput` from `gap-1` to `gap-2`

Pabl0cks commented on packages/nextjs/app/debug/_components/contract/TupleArray.tsx:
```diff
@@ -97,12 +97,12 @@ export const TupleArray = ({ abiTupleParameter, setParentForm, parentStateObject
+        <div className="ml-3 flex-col space-y-2 border-secondary/70 border-l-2 pl-4 collapse-content">
           {additionalInputs.map((additionalInput, additionalIndex) => (
             <div key={additionalIndex} className="space-y-1">
               <p className="m-0 ml-2 text-[0.95rem]">
```
Would look better deleting `ml-2` to align `tuple[n]` with its fields?

Pabl0cks reviewed (approved):
Small changes but big visual improvement! 🙌🔥

Left a couple of nitpick comments to save some future sleep problems haha. Feel free to ignore 👌

technophile-04 reviewed (approved):
Tysm !! Love this, it looks soo much better now!! 

Agree with all the review comments from Pablo 🙌

> As Shiv mentioned, maybe it's a good idea to have a label on the value input too.

Maybe something like this : 

![Screenshot 2024-02-15 at 9 54 52 AM](https://github.com/scaffold-eth/scaffold-eth-2/assets/80153681/23faa063-ae67-4d1a-84fb-c09f1120d06a)

Its just so that people can easily review while sending final transaction and don't loose what that last input is 

We can update : 
https://github.com/scaffold-eth/scaffold-eth-2/blob/a5eedbd783b49f5448dacce2a6287dfaf2359347/packages/nextjs/app/debug/_components/contract/WriteOnlyFunctionForm.tsx#L102

<details>
<summary>
With this : 
</summary>

```jsx
 {abiFunction.stateMutability === "payable" ? (
<div className="flex flex-col gap-1 w-full">
  <div className="flex items-center">
    <span className="text-xs font-medium mr-2 leading-none">value to send</span>
    <span className="block text-xs font-extralight leading-none">wei</span>
  </div>
  <IntegerInput
    value={txValue}
    onChange={updatedTxValue => {
      setDisplayedTxResult(undefined);
      setTxValue(updatedTxValue);
    }}
    placeholder="value (wei)"
  />
</div>
) : null}


```
<details>

carletex commented:
Thanks all for the review! Appreciate the nitpicking @Pabl0cks 

Since we are at it, fixed a few little things:

- More subtle color for the struct title
- Align internal padding with inputs
- Spacing between label / input or struct (1 => 1.5)
- Increase left padding of label (if not, because of the input/struct border radius, felt that was non-aligned)
- The difference of the radius (between inputs and structs) still bugs me, but cant have full radius on accordions.

![image](https://github.com/scaffold-eth/scaffold-eth-2/assets/2486142/873fad13-dc20-463c-87b9-92eebfa8d19e)

- Added the label to value fields (thanks for the snippet @technophile-04 !!)
-  I used "payable value". Not sure if the best option, but I wanted the "payable" word to be there.

![image](https://github.com/scaffold-eth/scaffold-eth-2/assets/2486142/44a7a147-0113-4c2f-ab7d-3ee5df5b86ad)

carletex commented:
Oh, forgot:

Better formatting for the tuple index indicator & buttons (+ / -) colors

![image](https://github.com/scaffold-eth/scaffold-eth-2/assets/2486142/f2c7f83a-ce2d-4a34-9779-da0263736f65)

technophile-04 commented:
> Since we are at it, fixed a few little things:

It just get better and better, love it !! 

---

Really sorry, just pushed a very small nitpick [7c766d0](https://github.com/scaffold-eth/scaffold-eth-2/pull/726/commits/7c766d0c5553837c684ce94574c02f146b2540a9), doesn't change any thing on light just affects the badge color in dark mode : 

| Before | After | 
| ------ | ------- | 
|  ![Screenshot 2024-02-15 at 2 45 01 PM](https://github.com/scaffold-eth/scaffold-eth-2/assets/80153681/5f131134-1413-49c5-be59-8a097ceb1cb0) | ![Screenshot 2024-02-15 at 2 40 06 PM](https://github.com/scaffold-eth/scaffold-eth-2/assets/80153681/5cd3186d-3824-4515-b386-401c13cdeea3) | 

Didn't change + / - btn color to `bg-base-300` since because of shadow they looks good 🙌

But incase if we don't like we can always update / iterate in future PR, Tysm !!

Pabl0cks commented:
Good job! We got a GREAT debug struct UI now 🚀🚀
